### DISCUSSION TITLE
Loading circle on mobile device

### DISCUSSION THREAD

**Discussion by mik-kun** [pretix/pretix#2724] [General]
Hi folks,
I noticed a spinning circle on mobiles devices (iPhone, tried Safari and Chrome), I'm attaching a screenshot of it.
Any ideas what that might be?

![IMG_7039-1](https://user-images.githubusercontent.com/107970619/180123906-d1e67cdf-e4fc-4866-9e5d-eccd20d7d0a8.PNG)

**Comment by raphaelm** [comment]
This sounds like old/incomplete css bring loaded. does it also happen on https://pretix.eu/demo/?style=list for your? 

if yes, what browser and os are your using?

if no, can you share the URL to where it happens?

**Reply by mik-kun** [reply]
Dear Raphel, I have just sent you the url to the email on your github profile.

- I'm not getting the spinning circle on your demo link.
- I already cleared my cache and tried with another iPhone - same result.

The Pretix server is a fresh installation with no customization except what you can do in the UI (change the color and header photo).

**Reply by raphaelm** [reply]
I received the link and can reproduce it, but I can't tell for sure what it is. I notice some CSP problems on the site which seem to be connected to cloudflare, but it more looks like an out-of-date-CSS issue. Is it possible you did an upgrade of pretix but did not perform the "updatestyles" step of the upgrade process?

**Reply by mik-kun** [reply]
Thank you for the check. It never crossed my mind it might be related to Cloudflare. I disabled their proxy for pretix and the spinning circle was gone. Re-enabled and not coming back. Solved for me.
As for your other question, I never did an upgrade, it's a brand new installation.

**Reply by mik-kun** [reply]
Have to make an edit on the above
after re-enabling the proxy I can reproduce the spinning circle. Seems to be an issue with Cloudflare + Pretix in combination 🤔

**Reply by mik-kun** [reply]
@raphaelm is that something you can test with a Cloudflare enabled domain? Or is there any rudimentary way to remove the footer/css to get rid of the spinning circle?
I would like to protect our IP but right now people would have to deal with that circle.

**Reply by raphaelm** [reply]
I have no cloudflare account and not really time to dig into this.

It *looked* like cloudflare was messing with the JavaScript in the page and injected JavaScript from a different origin, which was subsequently blocked by our content security policy header. This apparently lead to some off our JavaScript or CSS not executing properly. (The spinning circle is part of the image lightbox library we embed).

maybe it's possible in cloudflare to (a) not inject stuff or (b) adjust the CSP header.

**Reply by mik-kun** [reply]
@raphaelm Could you please point me in the right direction as where I can find the config for the CSP header? Which file or location I could search in. Couldn't find anything so far. Thank you.

**Reply by raphaelm** [reply]
We have csp_additional_header: https://docs.pretix.eu/en/latest/admin/config.html#pretix-settings

**Reply by mik-kun** [reply]
@raphaelm thank you. The syntax in the documentation isn't obvious for me. Is that a correct example below?
[pretix]
csp_additional_header=script-src https://mycdn.com

**Reply by mik-kun** [reply]
@raphaelm would be grateful for your confirmation

**Reply by wiffbi** [reply]
Yes, that looks right. You could even add more instructions, it just has to be a valid CSP-Header string as described e.g. at https://content-security-policy.com/

**Reply by mik-kun** [reply]
Unfortunately this doesn't make any difference, I tried various settings.
If someone uses Cloudflare and know's how to resolve this, please reach out.
